### PR TITLE
initial support for highlighting in multi crs situations

### PR DESCRIPTION
**Pull request by mbernasocchi**

### CODE REVIEW

File: src/featurelistmodelhighlight.cpp [hunk 1]
```diff
@@ -83,14 +91,18 @@ QSGNode* FeatureListModelHighlight::updatePaintNode( QSGNode* n, QQuickItem::Upd
       QModelIndex index = mModel->index( i, 0, QModelIndex() );
       const Feature& feature = mModel->data( index, FeatureListModel::FeatureRole ).value<Feature>();
 
+      QgsFeature feat( feature.qgsFeature() );
```
Comment (m-kuhn) [inline src/featurelistmodelhighlight.cpp]: No reason for a complete feature copy.

    QgsGeometry geom( feature.constGeometry() );
    geom.transform( transf );
    ....
    sn = new QgsSGGeometry( geom, mSelectionColor, mWidth );

should be ok.

Comment (m-kuhn) [inline src/featurelistmodelhighlight.cpp]: typo... updated above

File: src/featurelistextentcontroller.cpp [hunk 2]
```diff
@@ -38,7 +38,13 @@ void FeatureListExtentController::zoomToSelected() const
   if ( mModel && mSelection && mMapSettings )
   {
     Feature feat = mSelection->selectedFeature().value<Feature>();
-    QgsRectangle featureExtent = feat.qgsFeature().constGeometry()->boundingBox();
+
+    const QgsCoordinateReferenceSystem& layerCrs = feat.layer()->crs();
+    const QgsCoordinateTransform & transf = QgsCoordinateTransform(layerCrs, mMapSettings->crs()->crs());
```
Comment (m-kuhn) [inline src/featurelistextentcontroller.cpp]: QgsCoordinateTransform transf = QgsCoordinateTransform( feat.layer()->crs(), mMapSettings->crs()->crs() );

File: src/featurelistmodelhighlight.cpp [hunk 3]
```diff
@@ -76,6 +77,11 @@ QSGNode* FeatureListModelHighlight::updatePaintNode( QSGNode* n, QQuickItem::Upd
     int count = mModel->rowCount( QModelIndex() );
     QgsSGGeometry* sn = 0;
 
+    QModelIndex firstIndex = mModel->index( 0, 0, QModelIndex() );
+    const Feature& firstFeature = mModel->data( firstIndex, FeatureListModel::FeatureRole ).value<Feature>();
+    const QgsCoordinateReferenceSystem& layerCrs = firstFeature.layer()->crs();
+    const QgsCoordinateTransform & transf = QgsCoordinateTransform(layerCrs, mMapSettings->crs()->crs());
```
Comment (m-kuhn) [inline src/featurelistmodelhighlight.cpp]: Same as above

File: src/featurelistmodelhighlight.cpp [hunk 4]
```diff
@@ -76,6 +77,11 @@ QSGNode* FeatureListModelHighlight::updatePaintNode( QSGNode* n, QQuickItem::Upd
     int count = mModel->rowCount( QModelIndex() );
     QgsSGGeometry* sn = 0;
 
+    QModelIndex firstIndex = mModel->index( 0, 0, QModelIndex() );
+    const Feature& firstFeature = mModel->data( firstIndex, FeatureListModel::FeatureRole ).value<Feature>();
```
Comment (mbernasocchi) [inline src/featurelistmodelhighlight.cpp]: no

Comment (m-kuhn) [inline src/featurelistmodelhighlight.cpp]: Replace it anyway ;)

**Pull request merged**